paddle/fluid/framework/op_desc.cc

```diff
@@ -50,6 +50,28 @@ class CompileTimeInferShapeContext : public InferShapeContext {
   const std::vector<std::string> &Outputs(
       const std::string &name) const override;
 
+  void ShareDimAndLod(const std::string &in, const std::string &out,
```

Comment:
Why we should combine the ShareDimAndLod into one API?  The i, j is for the dims, the lod do not need them at all.

Reply:
> Why we should combine the ShareDimAndLod into one API?

There are two main reasons that combining the ShareLod and ShareDim, 
1. In most operator, the output var's dim and lod is the same with one of the input vars, for example, activation op and elementwise_op.
2. most logic of  ShareDim and ShareLod are the same.

> The i, j is for the dims, the lod do not need them at all.

i and j indicate which argument of `in` and `out`, they are not only for the dim.